### ISSUE TITLE
Balance totally incorrect on all of the accounts

### ISSUE THREAD

**Issue by optimistic5** [firefly-iii/firefly-iii#2715]
**Bug description**
I am running Firefly III version 4.8.1.1, and my problem is:
Balance totally incorrect on all of the accounts.
@JC5 I wrote you email, but apparently it goes to spam, please check more detail in my email.
I desperately need your help.

First month everything was smooth.
In the end of the mount I checked the balance of all the accounts and corrected it,
everything was accordingly to the real balance.

All October I created each transaction, and everything should be properly.
Just yesterday I find out that all UAH account shows incorrect balance.

![image](https://user-images.githubusercontent.com/25054050/66573394-486b3100-eb7b-11e9-910c-3dd4b60cc0ec.png)

account number 1 on screenshot.
It was 0 UAH at the end of the September, and I did only two small transaction.
It cannot be -2200UAH
You can see it here
![image](https://user-images.githubusercontent.com/25054050/66573505-76507580-eb7b-11e9-9004-1b605feaf653.png)

account number 2 on screenshot.
It was some small amount of the money (about 100 UAH) in the end of the month.
It cannot be -19361UAH, I didn't make such transaction.
You can see it here
![image](https://user-images.githubusercontent.com/25054050/66573587-96803480-eb7b-11e9-9cac-300ed19407c6.png)

number 3 on screenshot
You can see that I spend 3238UAH, it look like true.
![image](https://user-images.githubusercontent.com/25054050/66574072-4bb2ec80-eb7c-11e9-889c-f36f6976df75.png)

----------------
![image](https://user-images.githubusercontent.com/25054050/66574184-756c1380-eb7c-11e9-896e-aae0fd0742d9.png)
You can see that I overspend 3238UAH.
It can't be true, because I spend 3238, but my budget is 25000UAH
![image](https://user-images.githubusercontent.com/25054050/66574223-87e64d00-eb7c-11e9-8ecf-f82b15cb17e5.png)

It is a lot of strange numbers.
All balance is missed up.

I will provide you all info you need to investigate this issue.
Hope for your help.
Thank you a lot!

Debug information generated at 2019-10-10 16:28:27 Europe/Kiev for Firefly III version **4.8.1.1**.

| Variable | Content |
| --- | --- |
| FF version | 4.8.1.1 |
| FF API version | 0.10.2 |
| App environment | local |
| App debug mode | false |
| App cache driver | file |
| App logging | , daily |
| PHP version | 7.3.9 |
| Display errors | Off |
| Session start | 2019-10-01 00:00:00 |
| Session end | 2019-10-31 23:59:59 |
| Session first | 2019-09-01 00:00:00 |
| Error reporting | ALL errors |
| Host | Linux |
| Interface | apache2handler |
| UserID | 1 |
| Attempt at "en" | false |
| Attempt at "English" | false |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| DB drivers | mysql, pgsql, sqlite |
| Current driver | pgsql |
| Login provider |   |
| Storage disks | local-upload  |
| Using Sandstorm? | no |
| Is Sandstorm (.env) | false |
| Is Docker (.env) | false |
| bunq uses sandbox | false |
| Trusted proxies (.env) | ** |
| User agent | Mozilla/5.0 (X11; Linux x86_64; rv:69.0) Gecko/20100101 Firefox/69.0 |
| Loaded extensions | Core, date, libxml, openssl, pcre, sqlite3, zlib, ctype, curl, dom, fileinfo, filter, ftp, hash, iconv, json, mbstring, SPL, PDO, session, posix, Reflection, standard, SimpleXML, pdo_sqlite, Phar, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, apache2handler, bcmath, gd, intl, ldap, memcached, pdo_mysql, pdo_pgsql, sodium, Zend OPcache |

**Comment by optimistic5** [comment]
Just find out the issue #2694 
Few days ago I delete one of the account I not longer needed.
Are deletion of account with some transactions can make such problem?
What happens when account is deleted?

**Comment by JC5** [comment]
I saw your email. I can’t access my mail as easily as GitHub. Hence the delay in response. Sorry about that. 

If you delete an account, all associated transactions are also deleted. This includes transfers back and forth. This is mentioned in the dialog as well.

**Comment by optimistic5** [comment]
@JC5 no problem, thank you for your quick answer!

> If you delete an account, all associated transactions are also deleted. This includes transfers back and forth.

I saw the massage about it, but apparently it was not clear for me.
Didn't realize that it will became such a mess  :disappointed:  :cry: and all the accounts will be affected.

I'm agree with @Nik-vr, please make some feature for account that do not needed any more, archiving is a good idea.

I got a lesson, and go to study how to make automation DB backups.

To solve my issue it will be better to start from very begging, drop the DB and create everything one more time, yes?

**Comment by JC5** [comment]
Everything is still in the database. The field “deleted_at” will be filled in. 

If you make the field empty again data will appear again. Let me write detailed instructions later.

**Comment by optimistic5** [comment]
I found record `deleted_at`
![image](https://user-images.githubusercontent.com/25054050/66954234-a9f23a80-f068-11e9-856d-57fc80b261e2.png)
Removed it
```
UPDATE accounts SET deleted_at=NULL WHERE id=13;
UPDATE accounts SET deleted_at=NULL WHERE id=17;
```
![image](https://user-images.githubusercontent.com/25054050/66954361-f2115d00-f068-11e9-959c-efb93417b390.png)

In Firefly it look like:
![image](https://user-images.githubusercontent.com/25054050/66954478-25ec8280-f069-11e9-983f-fe33992db37d.png)

Balance didn't recovered.
For example budget still shows wrong number, it cannot be -7865UAH
![image](https://user-images.githubusercontent.com/25054050/66954597-61874c80-f069-11e9-9ee1-af101801d80d.png)

@JC5 , please, let me know
**What should I do, to recover my balance after restoring deleted account?**

**Comment by JC5** [comment]
Make sure you set the "deleted_at" column to NULL in the following tables:

* transactions
* transaction_groups
* transaction_journals

If possible, remove only the "deleted_at" values for entries on the moment you deleted the account. Something like this:

```sql
UPDATE <table> SET deleted_at = NULL WHERE deleted_at >= "2019-01-01 00:00:00 AND deleted_at "2019-01-02 00:00:00"
```

If you deleted your account on October 5th, it would be:

```sql
UPDATE <table> SET deleted_at = NULL WHERE deleted_at >= "2019-10-05 00:00:00 AND deleted_at "2019-10-06 00:00:00"
```

**Comment by optimistic5** [comment]
- accounts
- transactions
- transaction_groups
- transaction_journals
I did it. Above tables without any `deleted_at` in the date I deleted account.

![image](https://user-images.githubusercontent.com/25054050/67162725-892f2b00-f36f-11e9-8bf0-de8be03f8dc3.png)

But still I have incorrect balance on some of my default asset accounts and budget.

For example in budget I spend 20k from 25k
![image](https://user-images.githubusercontent.com/25054050/67162769-212d1480-f370-11e9-957f-068601dc81e6.png)
But on dashboard it shows that I overspend 20k
![image](https://user-images.githubusercontent.com/25054050/67162778-39049880-f370-11e9-8e62-75d9cc75ab30.png)

If nothing to update here - just let me know and will start from very beginning in the next mouth.

@JC5 thanks a lot for your help!

**Comment by JC5** [comment]
Sorry, @optimistic5, this fell of my plate. It looks like you undeleted everything correctly. I'm not sure what else I can do for you, sorry :(

**Comment by rnck** [comment]
Same issue here. I have a negative balance for my paypal account on the dashboard, on the default financial report and on the asset accounts page. In reality, the balance is positive. All other accounts show correct balances. I did not delete any asset accounts. All transactions for the paypal account are correct and the chart at accounts/show/5/all shows correct balance.

<img width="1653" alt="Bildschirmfoto 2019-10-29 um 09 41 12" src="https://user-images.githubusercontent.com/165663/67751110-cdd25a80-fa30-11e9-84e1-bd0c7c8f44bd.png">

But: At accounts/show/5 (current month) the balance drops below zero.

<img width="1660" alt="Bildschirmfoto 2019-10-29 um 09 41 27" src="https://user-images.githubusercontent.com/165663/67751151-dd51a380-fa30-11e9-999b-cb9c23d444d7.png">

Maybe this is related to #2780, because I deleted a budget. But I'm not sure, if I used the deleted budget on transactions related to that account.
